app/controllers/video_embed_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class VideoEmbedController < ApplicationController
+  def new
+    if rendering_context != "modal"
+      raise "Video embed only supported in modal context"
```

Comment:
Do you think there are ways we can avoid a raise here and/or what this should return ideally?

I know we've got precedent for raising when edition isn't in an expected state, but I'm not sure if we have one yet for non-modal content and would be good, if possible, to avoid having too many internal server error responses that are for client error.

I'm thinking it'd probably be either a 400 bad request or 412 precondition failed. What do you think?

Reply:
Sure - I think I'm still getting used to us moving away from exceptions by default. [This article](https://airbrake.io/blog/http-errors/400-bad-request) was also helpful to understand the decision making process.

As for the specific code, I think 412 is [reserved for specific headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412), but there seems to be [some debate around whether 400 is appropriate](https://stackoverflow.com/questions/19671317/400-bad-request-http-error-code-meaning). 

On the whole, a [422 response code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) seems the most accurate for this situation. What do you think?

Reply:
Ah shoot I should have read up more on precondition failed before suggesting - that probably explains why we never see it anywhere, despite the name feeling most appropriate.

So 422 is all wrapped in validations, particularly in Rails land, so I think we should avoid them - particularly as that'd make it non-distinct from issues. I think the closest precedent we already have in rails is with strong parameters which return a 400 response, for example if you do `params.require(:item)` and item isn't a parameter that would be 400.

Reply:
Ah, params makes a good precedent - updated.